### ISSUE TITLE
URL deep-linking breaks app (blank page) on iOS

### ISSUE THREAD

**Issue by AndreasGassmann** [ionic-team/cordova-plugin-ionic-webview#203]
**Expected behavior:**

The app should start normally when being started through a custom-URL scheme.

**Actual behavior:**

When using a url scheme to start an app that is not running (cold-start), the app then shows a blank screen. To make it worse, the blank screen persists, even if the app gets force closed.

Steps to reproduce:

1. Install app1
2. Start app1, everything works fine
3. Force close app1
4. Click on a url scheme to open app1 (cold start)
5. App shows a white screen.
6. Force-close app1 and reopen. White screen is still there. Even restarting the device doesn't help.

The only way to resolve the blank screen issue is to remove and reinstall the app.

We currently have not set up universal links, so I can't test that.

When opening the Safari Debugger, I can see that the `index.html` gets loaded, but it only contains:
```
<!DOCTYPE html>
<html><head><meta charset="utf-8"></head><body>
</body></html>
```

The XCode logs (when starting the app again after it broke):
```
2018-10-26 15:49:49.356528+0200 App1[2259:139119] Apache Cordova native platform version 4.5.3 is starting.
2018-10-26 15:49:49.356989+0200 App1[2259:139119] Multi-tasking -> Device: YES, App: YES
2018-10-26 15:49:49.365818+0200 App1[2259:139119] CDVWKWebViewEngine: trying to inject XHR polyfill
2018-10-26 15:49:49.671690+0200 App1[2259:139119] CDVWKWebViewEngine will reload WKWebView if required on resume
2018-10-26 15:49:49.671790+0200 App1[2259:139119] Using Ionic WKWebView
2018-10-26 15:49:49.672481+0200 App1[2259:139119] [CDVTimer][console] 0.123978ms
2018-10-26 15:49:49.672639+0200 App1[2259:139119] [CDVTimer][handleopenurl] 0.077009ms
2018-10-26 15:49:49.674504+0200 App1[2259:139119] [CDVTimer][intentandnavigationfilter] 1.837015ms
2018-10-26 15:49:49.674594+0200 App1[2259:139119] [CDVTimer][gesturehandler] 0.056982ms
2018-10-26 15:49:49.674682+0200 App1[2259:139119] CDVIonicKeyboard: resize mode 1
2018-10-26 15:49:49.675409+0200 App1[2259:139119] [CDVTimer][keyboard] 0.735998ms
2018-10-26 15:49:49.680010+0200 App1[2259:139119] [CDVTimer][splashscreen] 4.550934ms
2018-10-26 15:49:49.683888+0200 App1[2259:139119] [CDVTimer][statusbar] 3.825903ms
2018-10-26 15:49:49.684009+0200 App1[2259:139119] [CDVTimer][socialsharing] 0.084043ms
2018-10-26 15:49:49.684475+0200 App1[2259:139119] [CDVTimer][ionicdeeplinkplugin] 0.431061ms
2018-10-26 15:49:49.684505+0200 App1[2259:139119] [CDVTimer][TotalPluginStartup] 12.222052ms
```

Sadly I don't know how to get the native logs of the app start when cold-starting the app with a deep-link. If someone can explain how to do that and if it's helpful, I can post those logs as well.

It looks obvious that for some reason, a wrong index.html gets loaded and therefor the app never starts (correctly). But I'm not sure where this index.html is coming from. I first thought it might be this: https://github.com/ionic-team/cordova-plugin-ionic-webview/blob/570b1f7ef88ccf70d1c47fa445b921c34221018e/src/ios/GCDWebServer/Responses/GCDWebServerErrorResponse.m#L80, but that looks a little different.

**Additional Info:**

We currently have our app in the store with `cordova-plugin-ionic-webview` version `^1.2.1`. We noticed that when cold-starting the app via deep-links, only assets could get loaded that have been loaded previously when the app was running. But other assets (images, js files, etc) that were not used before, simply were not found and returned a 404.

We noticed this because some images, as well as a js file used in a webworker, were simply not loading sometimes. It took quite a while to figure out that it was due to the deep-linking cold start.

We would be perfectly happy with v1 if this issue was resolved, but I suppose you only plan on fixing bugs in v2?

**Info:**

```
Ionic:

   ionic (Ionic CLI)  : 4.1.2 (/Users/andreas/.nvm/versions/node/v10.3.0/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.1.9

Cordova:

   cordova (Cordova CLI) : 7.1.0
   Cordova Platforms     : android 7.1.0, ios 4.5.4
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 2.2.0, (and 11 other plugins)

System:

   Android SDK Tools : 26.1.1 (/Users/andreas/Library/Android/sdk)
   ios-deploy        : 1.9.2
   NodeJS            : v10.3.0 (/Users/andreas/.nvm/versions/node/v10.3.0/bin/node)
   npm               : 6.1.0
   OS                : macOS
   Xcode             : Xcode 10.0 Build version 10A255
```

```
com.lampa.startapp 6.1.6 "startApp"
cordova-plugin-add-swift-support 1.7.2 "AddSwiftSupport"
cordova-plugin-app-version 0.1.9 "AppVersion"
cordova-plugin-device 1.1.7 "Device"
cordova-plugin-inappbrowser 1.7.2 "InAppBrowser"
cordova-plugin-ionic-keyboard 2.1.3 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 2.2.0 "cordova-plugin-ionic-webview"
cordova-plugin-qrscanner 2.6.0 "QRScanner"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-plugin-x-socialsharing 5.4.1 "SocialSharing"
cordova-sqlite-storage 2.4.0 "Cordova sqlite storage plugin"
es6-promise-plugin 4.2.2 "Promise"
ionic-plugin-deeplinks 1.0.17 "Ionic Deeplink Plugin"
```

**Comment by AndreasGassmann** [comment]
One thing that I just noticed today: If I open the Safari Debugger while the app shows the blank page and refresh the app there, it loads correctly. So it seems to me that the deep-linking somehow "bypasses" certain startup mechanisms, which causes the app to load before the web server is ready with the actual content? I'm still not sure where that blank HTML page is coming from though.

**Comment by MoacirSchmidt** [comment]
I've noticed similar problem (after SEVERAL hours of deep debugging)

Deep links stopped working after upgrading from "cordova-plugin-ionic-webview": "^1.2.1".

I use Google Firebase Authentication with Cordova (https://firebase.google.com/docs/auth/web/cordova) and they relies on cordova-universal-links-plugin-fix (I also tried cordova-universal-links-plugin) to implement deep links. 

After upgrading to cordova-plugin-ionic-webview" deep links stopped working. I had to downgrade to 1.2.1 where everything works ok!
